client/client_test.go

```diff
@@ -60,14 +60,15 @@ func setupCloudproberAndClient(t *testing.T) *CloudproberClient {
 func teardownCloudproberAndClient(t *testing.T, client *CloudproberClient) {
 	// TODO(evanSpendlove): Find a safe way to close down Cloudprober and restart it.
 	// Check if the probe was added correctly
+	ctx := context.Background()
 	t.Helper()
-	probes, err := client.ListProbes(context.Background())
+	probes, err := client.ListProbes(ctx)
 	if err != nil {
 		t.Fatalf("ListProbes() failed during teardown: %v", err)
 	}
 
 	for _, p := range probes {
-		if err = client.RemoveProbe(context.Background(), p.GetName()); err != nil {
+		if err = client.RemoveProbe(ctx, p.GetName()); err != nil {
 			t.Errorf("RemoveProbe() failed, expected error: %v, got %v", nil, err)
```

Comment:
Fatalf
Also, simplify the error message, as this is not a test expectation:

t.Fatalf("RemoveProbe failed: %v", err)

Reply:
I've changed this accordingly.